### ISSUE TITLE
PSR12.Properties.ConstantVisibility false positive when using public final const syntax

### ISSUE THREAD

**Issue by FlintMayers** [squizlabs/PHP_CodeSniffer#3526]
**Describe the bug**
False positive found with the example below
**Code sample**
```php
class SampleEnum
{
    public final const SAMPLE_CONST = 'SAMPLE';
}

```

**To reproduce**
1. Create a file called `SampleEnum.php` with the code sample above...
2. Run `phpcs SampleEnum.php ...`
3. See error message displayed
```
Visibility must be declared on all constants if your project supports PHP 7.1 or later
    |         | (PSR12.Properties.ConstantVisibility.NotFound)
```

**Expected behavior**
False positive. It shouldn't show a warning.

**Versions (please complete the following information):**
 - OS: Linux
 - PHP: 8.1
 - PHPCS: 3.6.2
 - Standard: PSR12

**Comment by jrfnl** [comment]
@FlintMayers Thanks for reporting this. PHP 8.1 support is still being worked on - see #3479.

Even so, I have confirmed the issue and pulled a fix in #3527. Testing appreciated.

**Comment by gsherwood** [comment]
PR worked for me and has been merged. Thanks for reporting this.
